typings/vscode-tasks.d.ts

```diff
@@ -0,0 +1,353 @@
+        /**
+         * Additional information used to detect when a background task (like a watching task in Gulp)
+         * is active.
+         */
+        background?: BackgroundMatcher;
+    }
+
+    /**
+     * A description to track the start and end of a background task.
+     */
+    export interface BackgroundMatcher {
+
+        /**
+         * If set to true the watcher is in active mode when the task
+         * starts. This is equals of issuing a line that matches the
+         * beginPattern.
+         */
+        activeOnStart?: boolean;
+
+        /**
+         * If matched in the output the start of a background task is signaled.
+         */
+        beginsPattern?: string;
+
+        /**
+         * If matched in the output the end of a background task is signaled.
+         */
+        endsPattern?: string;
+    }
+
+    export interface ProblemPattern {
+
+        /**
+         * The regular expression to find a problem in the console output of an
+         * executed task.
+         */
+        regexp: string;
+
+        /**
+         * The match group index of the filename.
+         */
+        file: number;
+
+        /**
+         * The match group index of the problems's location. Valid location
+         * patterns are: (line), (line,column) and (startLine,startColumn,endLine,endColumn).
+         * If omitted the line and column properties are used.
+         */
+        location?: number;
+
+        /**
+         * The match group index of the problem's line in the source file.
+         * Can only be omitted if location is specified.
+         */
+        line?: number;
+
+        /**
+         * The match group index of the problem's column in the source file.
+         */
+        column?: number;
+
+        /**
+         * The match group index of the problem's end line in the source file.
+         *
+         * Defaults to undefined. No end line is captured.
+         */
+        endLine?: number;
+
+        /**
+         * The match group index of the problem's end column in the source file.
+         *
+         * Defaults to undefined. No end column is captured.
+         */
+        endColumn?: number;
+
+        /**
+         * The match group index of the problem's severity.
+         *
+         * Defaults to undefined. In this case the problem matcher's severity
+         * is used.
+         */
+        severity?: number;
+
+        /**
+         * The match group index of the problem's code.
+         *
+         * Defaults to undefined. No code is captured.
+         */
+        code?: number;
+
+        /**
+         * The match group index of the message. Defaults to 0.
+         */
+        message: number;
+
+        /**
+         * Specifies if the last pattern in a multi line problem matcher should
+         * loop as long as it does match a line consequently. Only valid on the
+         * last problem pattern in a multi line problem matcher.
+         */
+        loop?: boolean;
+    }
+}
```

Comment:
What's this third-party file and why is it added to the project? Can't this be referenced as a dependency?